### PR TITLE
Added strings for the share dialog

### PR DESCRIPTION
**Pull request by Jaiganeshkumaran**
- Now when you share a single item, the item name will be mentioned.
- Sharing multiple items will show how many items are being shared.
- Share menu item is now disable on unsupported devices (e.g. Xbox). This is the first step in making Files UWP truly universal across Windows devices.
- All strings related to the share dialog are localized.

### CODE REVIEW

File: Files/Interacts/Interaction.cs [hunk 1]
```diff
@@ -651,12 +651,23 @@ private async void Manager_DataRequested(DataTransferManager sender, DataRequest
...
             {
-                dataRequest.FailWithDisplayText("Could not access file(s) for sharing");
+                dataRequest.FailWithDisplayText(ResourceController.GetTranslation("ShareDialogFailMessage"));
                 dataRequestDeferral.Complete();
                 return;
             }
+            else
+            {
+                dataRequest.Data.Properties.Title = string.Format(ResourceController.GetTranslation("ShareDialogTitle"), items.Count) + 
+                    $" {ResourceController.GetTranslation("ItemsCount.Text")}";
```
Comment (tsvietOK) [inline Files/Interacts/Interaction.cs:668]:
```suggestion
                    $"{ResourceController.GetTranslation("ItemsCount.Text")}";
```

**Comment by Jaiganeshkumaran** [comment]
@tsvietOK No space is required otherwise it will appear like 'Sharing 1items'.

File: Files/MultilingualResources/Files.de-DE.xlf [hunk 2]
```diff
@@ -1139,6 +1139,22 @@
...
+        <trans-unit id="ShareDialogMultipleItemsDescription" translate="yes" xml:space="preserve">
+          <source>The items you've selected will be shared</source>
+          <target state="new">The items you've selected will be shared</target>
+        </trans-unit>
+        <trans-unit id="ShareDialogSingleItemDescription" translate="yes" xml:space="preserve">
+          <source>The selected item will be shared</source>
+          <target state="new">The selected item will be shared</target>
+        </trans-unit>
+        <trans-unit id="ShareDialogTitle" translate="yes" xml:space="preserve">
+          <source>Sharing {0}</source>
```
Comment (tsvietOK) [inline Files/MultilingualResources/Files.de-DE.xlf:1155]: It is better to use `Sharing {0} {1}` and `string.Format(ResourceController.GetTranslation("ShareDialogTitle"), items.Count, ResourceController.GetTranslation("ItemsCount.Text"))`

Comment (Jaiganeshkumaran) [reply Files/MultilingualResources/Files.de-DE.xlf:1155]: Done

Comment (tsvietOK) [reply Files/MultilingualResources/Files.de-DE.xlf:1155]: Please, update translation files too.

**Pull request merged**